pkg/cloud/gcp/actuators/util/gcp_tags_labels.go

```diff
@@ -119,3 +130,100 @@ func GetLabelsList(userLabelsAllowed bool, client controllerclient.Client, machi
 
 	return labels, nil
 }
+
+// getInfraResourceTagsList returns the user-defined tags present in the
+// status sub-resource of Infrastructure.
+func getInfraResourceTagsList(platformStatus *configv1.PlatformStatus) []machinev1.ResourceManagerTag {
+	if platformStatus != nil && platformStatus.GCP != nil && platformStatus.GCP.ResourceTags != nil {
+		tags := make([]machinev1.ResourceManagerTag, len(platformStatus.GCP.ResourceTags))
+		for i, tag := range platformStatus.GCP.ResourceTags {
+			tags[i] = machinev1.ResourceManagerTag{
+				ParentID: tag.ParentID,
+				Key:      tag.Key,
+				Value:    tag.Value,
+			}
+		}
+		return tags
+	}
+	return nil
+}
```

Comment:
I think this would be simpler to follow if we negated this and removed some of the indentation
```suggestion
	if platformStatus == nil || platformStatus.GCP == nil || platformStatus.GCP.ResourceTags == nil {
		return nil
	}
	
	tags := make([]machinev1.ResourceManagerTag, len(platformStatus.GCP.ResourceTags))
	
	for i, tag := range platformStatus.GCP.ResourceTags {
		tags[i] = machinev1.ResourceManagerTag{
			ParentID: tag.ParentID,
			Key:      tag.Key,
			Value:    tag.Value,
		}
	}
	
	return tags
	
}
```
Takes up more lines, but in general is a more accepted pattern